py_neuromodulation/nm_stream_abc.py

```diff
@@ -155,6 +156,9 @@ def save_after_stream(
 
         self.save_sidecar(out_path_root, folder_name)
 
+        if not save_csv:
+            feature_arr = feature_arr.head()
```

Comment:
I think it's better to not load the whole dataframe and subset it here, we can read just the first line from the SQLite database and pass it to this function